lib/private/DB/AdapterMySQL.php

```diff
@@ -39,16 +39,16 @@ public function unlockTable() {
 	}
 
 	public function fixupStatement($statement) {
-		$statement = str_replace(' ILIKE ', ' COLLATE ' . $this->getCharset() . '_general_ci LIKE ', $statement);
+		$statement = str_replace(' ILIKE ', ' COLLATE ' . $this->getCollation() . ' LIKE ', $statement);
 		return $statement;
 	}
 
-	protected function getCharset() {
-		if (!$this->charset) {
+	protected function getCollation(): string {
+		if (!$this->collation) {
 			$params = $this->conn->getParams();
-			$this->charset = isset($params['charset']) ? $params['charset'] : 'utf8';
+			$this->collation = $params['collation'] ?? ($params['charset'] ?? 'utf8') . '_general_ci';
```

Comment:
Just for the record, ` $params['collation']` is not being appended ` . '_general_ci'`

Maybe we can add brackets for readability?

```suggestion
			$this->collation = $params['collation'] ?? (($params['charset'] ?? 'utf8') . '_general_ci');
```

Reply:
> Just for the record, $params['collation'] is not being appended . '_general_ci'

Yes, intentionally.

```
php > print($params['collation'] ?? ($params['charset'] ?? 'utf8') . '_general_ci');
utf8_general_ci

php > $params = ['charset' => 'utf8mb4'];
php > print($params['collation'] ?? ($params['charset'] ?? 'utf8') . '_general_ci');
utf8mb4_general_ci

php > $params['collation'] = 'utf8mb4_0900_as_ci';
php > print($params['collation'] ?? ($params['charset'] ?? 'utf8') . '_general_ci');
utf8mb4_0900_as_ci
```

> Maybe we can add brackets for readability?

sure